### PR TITLE
fix: null enum

### PR DESCRIPTION
**Pull request by ouertani**
**Summary:**  
_Briefly describe the changes made in this PR._

inner null enum failed to resolved.

**Issue Reference(s):**  
Fixes #... _(Replace "..." with the issue number)_

**Build & Testing:**

- [ X] I ran `cargo test` successfully.
- [ X] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ X] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [X ] I have performed a self-review of my code.
- [ ] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

**Comment by ouertani** [comment]
tried to fix with https://github.com/async-graphql/async-graphql/pull/1551 but actually this PR should resolve the issue.

File: assets/star-our-repo.gif
Comment (ssddOnTop) [inline assets/star-our-repo.gif:1]: can you revert this change

Comment (ouertani) [reply assets/star-our-repo.gif:1]: Done

File: src/core/blueprint/into_schema.rs [hunk 1]
```diff
@@ -107,9 +107,10 @@ fn to_type(def: &Definition) -> dynamic::Type {
                                 let ctx: ResolverContext = ctx.into();
                                 let ctx = EvalContext::new(req_ctx, &ctx);
 
-                                FieldFuture::from_value(
-                                    ctx.path_value(&[field_name]).map(|a| a.into_owned()),
-                                )
+                                match ctx.path_value(&[field_name]).map(|a| a.into_owned()) {
+                                    Some(ConstValue::Null) => FieldFuture::Value(FieldValue::NONE),
+                                    a => FieldFuture::from_value(a),
+                                }
```
Comment (ssddOnTop) [inline src/core/blueprint/into_schema.rs:113]: maybe we can not do this here, there might be some cases where we need to throw an error in the case of Null

Can you explain why this change makes more sense to you then the changes you made at async-graphql?

Comment (ouertani) [reply src/core/blueprint/into_schema.rs:113]: I have a use case where an internal type ( which is enum is null ) with the graphql directive. the reoslution always fails with error. if you check the test case it will fails without such modification

Comment (ouertani) [reply src/core/blueprint/into_schema.rs:113]:
```
-old snapshot
+new results
────────────┬────────────────────────────────────────────────────────────────────
    2     2 │   "headers": {
    3     3 │     "content-type": "application/json"
    4     4 │   },
    5     5 │   "body": {
    6       │-    "data": {
    7       │-      "post": {
    8       │-        "severity": {
    9       │-          "type": null
   10       │-        }
          6 │+    "data": null,
          7 │+    "errors": [
          8 │+      {
          9 │+        "message": "internal: invalid item for enum \"EnumType\"",
         10 │+        "locations": [
         11 │+          {
         12 │+            "line": 1,
         13 │+            "column": 27
         14 │+          }
         15 │+        ],
         16 │+        "path": [
         17 │+          "post",
         18 │+          "severity",
         19 │+          "type"
         20 │+        ]
   11    21 │       }
   12       │-    }
         22 │+    ]
   13    23 │   }
   14    24 │ }
```

Comment (ouertani) [reply src/core/blueprint/into_schema.rs:113]: @ssddOnTop I added another test to make sure that it will fail with exception when it is mandatory

File: tests/execution/graphql-datasource-with-empty-enum.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+type WithOptEnum {
+  type: EnumType
+}
+
+type Post {
+  severity: WithOptEnum!
+}
+
+type Query {
+  post: Post @graphQL(baseURL: "http://upstream/graphql", name: "post")
```
Comment (tusharmath) [inline tests/execution/graphql-datasource-with-empty-enum.md:22]: We can write this test using `@expr` https://tailcall.run/docs/tailcall-dsl-graphql-custom-directives/#static

**Comment by tusharmath** [comment]
Thanks @ouertani! Overall PR looks great, just a tiny little feedback about the test.

File: tests/execution/graphql-datasource-with-empty-enum.md (hunk 2 above)
Comment (ouertani) [reply tests/execution/graphql-datasource-with-empty-enum.md]: @tusharmath updated with `@expr`

**Review by tusharmath — APPROVED** [review]
